### PR TITLE
Cli switch register

### PR DESCRIPTION
**Pull request by SahilTikale**
Adding cli call for the api call Switch Register. 
I modified the mockswitch driver to include the hostname, username and password.

Most switches take at least these parameters. 

It helped me test the cli.

### CODE REVIEW

**Comment by SahilTikale** [comment]
Fixed the tests that my commit had broken.

File: docs/USING.rst [hunk 1]
```diff
@@ -50,3 +50,40 @@ to hard-disk booting the installed system.
 This is, as the filepath states, merely an example of how you might deploy to
 physical nodes.  Existing deployment systems such as Canonical's MAAS have also
 been run succesfully.
+
+Usage examples 
+====================
+
+Included herwith are some examples
```
Comment (knikolla) [inline docs/USING.rst]: grammar: herewith

File: haas/api.py [hunk 2]
```diff
@@ -143,8 +143,10 @@ def project_detach_node(project, node):
     for nic in node.nics:
         if nic.current_action is not None:
             raise BlockedError("Node has pending network actions")
-    node.stop_console()
-    node.delete_console()
+#    node.stop_console()
+#    node.delete_console()
```
Comment (knikolla) [inline haas/api.py]: Is there any reason why we're leaving the old code commented out?

**Comment by SahilTikale** [comment]
Ignore this pull request. It is mistakenly based on a non-master branch.
Instead review #530

**Pull request closed without merging**